meson.build

```diff
@@ -57,88 +70,73 @@ else
     '-Wno-mismatched-tags'])
 endif
 
-#Link with stdc++fs if it is supplied by the compiler (gcc v8 or older, clang v10 or older)
-#This is currently necessary on semaphore CI, which is running gcc8 max.
-#Has_Link_Argument is only supported in meson 0.46+
+# Link with stdc++fs if it is supplied by the compiler (gcc v8 or older, clang v10 or older)
+# This is currently necessary on semaphore CI, which is running gcc8 max.
+# Has_Link_Argument is only supported in meson 0.46+
 if compiler.has_link_argument('-lstdc++fs')
   add_project_link_arguments(['-lstdc++fs'], language: 'cpp')
 endif
 
-#Dependencies
-libxmlpp = dependency('libxml++-3.0', required : false)
+# Dependencies
+nulldependency = dependency('', required: false)
+
+libxmlpp = dependency('libxml++-3.0', required: false)
 if libxmlpp.found()
   conf.set('LIBXMLPP_VERSION_3_0', 1)
   conf.set('LIBXMLPP_VERSION_2_6', 0)
 else
-  libxmlpp = dependency('libxml++-2.6')
+  libxmlpp = dependency('libxml++-2.6', required: true)
   conf.set('LIBXMLPP_VERSION_2_6', 1)
   conf.set('LIBXMLPP_VERSION_3_0', 0)
+  warning('libxml++-2.6 is deprecated.  Use a more recent version if possible.')
 endif
 
-#OpenCV may be referred to as opencv or opencv4
+# OpenCV may be referred to as opencv or opencv4
+opencv = nulldependency
 if get_option('usewebcam')
-  message('OpenCV build requested')
   opencv = dependency('opencv', required: false)
-  if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-  else
-    opencv = dependency('opencv4', method : 'pkg-config', required: false)
-    if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-    else
-      error('OpenCV not found but webcam build requested')
-    endif
+  if not opencv.found()
+    opencv = dependency('opencv4', required: true)
   endif
-else
-  opencv=dependency('',required: false)
 endif
 
-#Portmidi is maybe not found in pkg-config, use find library as an alternative
+# Portmidi is maybe not found in pkg-config, use find library as an alternative,
+# with both names libportmidi (on ubuntu) and portmidi (on OSX)
+portmidi = nulldependency
 if get_option('usemididrum')
-  message('Portmidi build requested')
   portmidi = dependency('portmidi', required: false)
-  if portmidi.found()
-    add_project_arguments('-DUSE_PORTMIDI', language: 'cpp')
-  else
+  if not portmidi.found()
     portmidi = compiler.find_library('libportmidi', required: false)
```

Comment:
Please add `/opt/local/lib` as an additional path to look libportmidi under.

Reply:
Adding a path is not additional to the default search path, but replaces the path altogether.
I guess we could add a bit of logic around that if this is the only library installed in this location.
To be sure: could you also try with the scripts in this PR, as I seem to recall there was an outstanding parameter forcing pkg-config search for this library.

Reply:
I was gonna ask whether you're sure about this but I just checked the meson code and it appears you're correct.
I did, and tried other variations too. None of them work because portmidi doesn't provide a `.pc` file.

In the end, what meson is doing is trying to compile a simple program using `-L/usr/local/lib -llibportmidi`. One solution would be to check whether we're on darwin and if so add both `/usr/local/lib` and `/opt/local/lib` (which still leaves out the case of people who might install macports to a different prefix but frankly it's not worth it at this point)

Reply:
I believe you may have used a previous version of the script.

On homebrew, portmidi is found using this name:
```
Run-time dependency portmidi found: NO (tried pkgconfig, framework and cmake)
Library libportmidi found: NO
Library portmidi found: YES
```
So with 'portmidi' it should be fine, after it is installed with brew install portmidi
Could you give it another try before we try to fix it in a different way ?